src/components/views/directory/NetworkDropdown.js

```diff
@@ -26,19 +27,27 @@ export default class NetworkDropdown extends React.Component {
         this.onInputClick = this.onInputClick.bind(this);
         this.onRootClick = this.onRootClick.bind(this);
         this.onDocumentClick = this.onDocumentClick.bind(this);
-        this.onNetworkClick = this.onNetworkClick.bind(this);
+        this.onMenuOptionClick = this.onMenuOptionClick.bind(this);
+        this.onInputKeyUp = this.onInputKeyUp.bind(this);
         this.collectRoot = this.collectRoot.bind(this);
+        this.collectInputTextBox = this.collectInputTextBox.bind(this);
+
+        this.inputTextBox = null;
 
         this.state = {
             expanded: false,
-            selectedNetwork: null,
+            selectedServer: MatrixClientPeg.getHomeServerName(),
+            selectedNetwork: '_matrix',
         };
     }
 
     componentWillMount() {
         // Listen for all clicks on the document so we can close the
         // menu when the user clicks somewhere else
         document.addEventListener('click', this.onDocumentClick, false);
+
+        // fire this now so the defaults can be set up
+        this.props.onOptionChange(this.state.selectedServer, this.state.selectedNetwork);
```

Comment:
so... RoomDirectory renders, which mounts a NetworkDropdown, which changes the state of RoomDirectory, which makes RoomDirectory render...

I'm slightly surprised react doesn't bitch about it, but I guess if it works, it's ok.